### PR TITLE
Implement read() in terms of read(buffer, len)

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by peterbarker** [comment]
Serial-pass-through still works (tested on CubeBlack with USB+usb-tty adapter from telem1)

**Comment by peterbarker** [comment]
slcan pass-through still works fine

**Comment by peterbarker** [comment]
linux (navio2) still works just fine

File: libraries/AP_HAL/utility/BetterStream.h [hunk 1]
```diff
@@ -42,7 +42,13 @@ class AP_HAL::BetterStream {
 
     /* return value for read():
      * -1 if nothing available, uint8_t value otherwise. */
-    virtual int16_t read() = 0;
+    int16_t read() {
```
Comment (tridge) [inline libraries/AP_HAL/utility/BetterStream.h:45]: should be in cpp

**Review by tridge — CHANGES_REQUESTED** [review]
needs to be smaller or faster :-)

**Pull request closed without merging**